src/plugins/plugin-http.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+var shimmer = require('shimmer');
+var url = require('url');
+var isString = require('is').string;
+var merge = require('lodash.merge');
+var httpAgent = require('_http_agent');
+
+function requestWrap(api, request) {
+  var labels = api.labels;
+  function setTraceHeader(parsedOptions, context) {
+    if (context) {
+      return merge(parsedOptions, {
+        headers: {
+          [TraceLabels.TRACE_CONTEXT_HEADER_NAME]: context
```

Comment:
I would suggest ensuring that the HTTP hook test passes locally, since that would easily catch minor things like this.